C4SGSvc/src/main/java/org/c4sg/controller/UserController.java

```diff
@@ -81,4 +87,60 @@ public void deleteUser(@ApiParam(value = "User id to delete", required = true)
                                 @RequestParam(required = false) String lastName) {
         return userService.search(userName, firstName, lastName);
     }
+    
+    @RequestMapping(value = "/{userId}/uploadAvatar", method = RequestMethod.POST)
+    @ApiOperation(value = "Add new upload Avatar")
+    public String uploadAvatar(@ApiParam(value = "user Id", required = true)
+                             @PathVariable Integer userId,
+                             @ApiParam(value = "Request Body", required = true)
+                             @RequestBody String avatarFileContent) {
+    	FileOutputStream fos=null;
+        try {
+            byte[] imageByte = Base64.decodeBase64(avatarFileContent);
+            File directory = new File(ApplicationConstants.AVATAR_UPLOAD_DIRECTORY);
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            File f = new File(userService.getAvatarUploadPath(userId));
+            fos = new FileOutputStream(f);
```

Comment:
Change to use Java 7 approach, follow this: https://www.mkyong.com/java/how-to-read-file-in-java-fileinputstream/